### DISCUSSION TITLE
how get the property data by sql or Presentation

### DISCUSSION THREAD

**Discussion by paprophet** [iTwin/itwinjs-core#5427] [Q&A]
the element property in microstation have many element properties,
but in imodel viewer show just some element properties
how show the properties data or get the properties data by sql or  Presentation RULESET.

![6660](https://user-images.githubusercontent.com/18022720/233389118-a0adb881-fb38-4b26-ae99-1235f0f9146c.png)
![66666](https://user-images.githubusercontent.com/18022720/233389132-40d53023-9a88-47b4-9d02-40acab8abf9a.png)

**Comment by ColinKerr** [comment]
The default rule set should show all properties that were brought over into the iModel.  

As a first step try changing the selection scope from Element to Top Assembly
![image](https://user-images.githubusercontent.com/33028649/233460701-c1512a81-33ae-49c5-849d-6fc288806f1b.png)

It is possible that the 'missing data' is actually on a parent element that is non-graphical.

If that doesn't work it may be that the proper workspace is needed to load the dgn data and the connector does not have the correct workspace when synchronizing the data.

If none of this helps please provide more information about the data on the dgn side and how you are bringing the data over into the iModel.  For example is the missing data using Item Types, a schema imported into the dgn, referenced externally or something else.

**Reply by paprophet** [reply]
i  change the dgn  to imodel by   the iTwin Snapshot .
"For example is the missing data using Item Types, a schema imported into the dgn, referenced externally or something else."
please tell me how do?
and i can give you the dgn and imodel file ,if you tell me the email (my email  173988573@qq.com )

**Reply by ColinKerr** [reply]
@jffmarker

**Reply by jffmarker** [reply]
Yeah, I think we'll need to see how your data is formed to answer questions. If you can send a link to share the DGN to 05-halons-scrawl@icloud.com, I can provide it to the Mstn Connector team to help.

**Reply by jffmarker** [reply]
We received the DGN data, and will keep investigating. So far, the DGN data seems to be special, because the data are stored in Intrinsic ECXD Instances, and are not getting deserialized into ItemTypes correctly for the Connector to process.

**Reply by jffmarker** [reply]
@paprophet, this DGN file seems to have been generated as an export from Revit. We'd recommend using the original RVT file through the Revit Connector if possible, you'll get better fidelity that way.

The properties are not getting converted because we cannot find or process the Rfa schema that the Item Types use... but you've already lost some intelligence by going from RVT to DGN this way, hence why direct RVT conversion would be better.

CC: @c-macdonald @ColinKerr
